web/src/components/tasks/NaicsCodeInput.tsx

```diff
@@ -141,9 +144,14 @@ export const NaicsCodeInput = (props: Props): ReactElement => {
               name={"naics-radio-group"}
               value={naicsCode}
               onChange={(event: React.ChangeEvent<{ name?: string; value: string }>): void => {
-                setNaicsCode(event.target.value);
                 if (displayInput) setDisplayInput(false);
-                setInProgress();
+                if (props.isAuthenticated === IsAuthenticated.TRUE) {
+                  setNaicsCode(event.target.value);
+                  setInProgress();
+                } else {
+                  setShowNeedsAccountModal(true);
+                  setDisplayInput(false);
```

Comment:
What's the purpose of this line?

Reply:
I set `setDisplayInput(false)` so that when we click the NAICS code input radio button as a guest, the input field does not display. Otherwise, it does display and even though the modal appears, the task progress changes from NOT STARTED to COMPLETED.

Reply:
Similar to what we were trying to accomplish with the task status, we want to prevent a value from being saved, not undo a change to achieve the desired outcome.

I don't know if we necessarily need `setDisplayInput(false)`. If we do need it, then we should prevent that value from getting set in the first place.

Reply:
This is part of what's causing one of two issues I'm seeing. See my main comment for details

Reply:
@AnJuHyppolite Can this get removed?

Reply:
^^^